test/zstream_test.exs

```diff
@@ -349,4 +429,55 @@ defmodule ZstreamTest do
     Logger.debug("Total memory: #{total}")
     assert total < 150
   end
+
+  defp verify_aes_password(entries, password) do
+    verify_aes_password_with_options(entries, password)
+    verify_aes_password_with_options(entries, password, zip64: true)
+  end
+
+  defp verify_aes_password_with_options(entries, password, options \\ []) do
+    Temp.track!()
+    path = Temp.path!(%{suffix: ".zip"})
+
+    Zstream.zip(entries, options)
+    |> Stream.into(File.stream!(path))
+    |> Stream.run()
+
+    # Use 7zz to verify the AES encrypted archive
+    {response, exit_code} = System.cmd("7zz", ["l", path])
+    Logger.debug("7zz list output: #{response}")
+    assert exit_code == 0
+
+    # Test the archive integrity and password
+    {response, exit_code} = System.cmd("7zz", ["t", "-p#{password}", path])
+    Logger.debug("7zz test output: #{response}")
+    assert exit_code == 0
+
+    # Extract and verify file contents
+    temp_dir = Temp.mkdir!()
+
+    {response, exit_code} =
+      System.cmd("7zz", ["x", "-p#{password}", "-o#{temp_dir}", path, "-y"])
+
+    Logger.debug("7zz extract output: #{response}")
+    assert exit_code == 0
+
+    # Verify extracted file contents match original
+    entries = Enum.reject(entries, fn e -> String.ends_with?(e.name, "/") end)
+
+    Enum.each(entries, fn entry ->
+      extracted_path = Path.join(temp_dir, entry.name)
+
+      if File.exists?(extracted_path) do
```

Comment:
Do we expect any file to be missing? we would just do `assert File.exists?(extracted_path)`?